java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java

```diff
@@ -0,0 +1,95 @@
+package com.ibm.streamsx.topology.internal.context;
+
+import static com.ibm.streamsx.topology.context.ContextProperties.KEEP_ARTIFACTS;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import com.ibm.json.java.JSONObject;
+import com.ibm.streamsx.topology.Topology;
+import com.ibm.streamsx.topology.context.ContextProperties;
+import com.ibm.streamsx.topology.internal.process.CompletedFuture;
+
+public class ZippedToolkitStreamsContext extends ToolkitStreamsContext {
+	
+	@Override
+	public Future<File> submit(Topology app, Map<String, Object> config) throws Exception {        
+        File toolkitRoot = super.submit(app, config).get();
+        Path zipOutPath = pack(toolkitRoot.toPath());
+        if (!(config.containsKey(KEEP_ARTIFACTS) && (boolean)config.get(KEEP_ARTIFACTS)))
+        	delete(toolkitRoot.toPath());
+        return new CompletedFuture<File>(zipOutPath.toFile());
+	}
+	
+	@Override
+	public Future<File> submit(JSONObject submission) throws Exception {
+        File toolkitRoot = super.submit(submission).get();
+        Path zipOutPath = pack(toolkitRoot.toPath());
+        
+        JSONObject deployInfo = (JSONObject)  submission.get(SUBMISSION_DEPLOY);
+        if (!(deployInfo.containsKey(KEEP_ARTIFACTS) && (boolean)deployInfo.get(KEEP_ARTIFACTS)))
+        	delete(toolkitRoot.toPath());
+        return new CompletedFuture<File>(zipOutPath.toFile());
+	}
+	
+	// Deletes a file or directory, even if not empty. Returns false if
+	// it could not be deleted.
+	public static boolean delete(final Path folder) {
```

Comment:
Since we already delete a toolkit, we must have delete directory functionality already